### ISSUE TITLE
tsql - parse error when turning multiple settings on for procedure.

### ISSUE THREAD

**Issue by ddbtrmatic** [sqlfluff/sqlfluff#6197]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Encountering a parsing error when setting multiple options in a single statement.

```sql
CREATE OR ALTER PROCEDURE dbo.DoSomething
AS
BEGIN
    SET NOCOUNT, XACT_ABORT ON;
END
GO
```

### Expected Behaviour

`SET NOCOUNT, XACT_ABORT ON;` is valid tsql and should be parsed..

### Observed Behaviour

```
== [file.sql] FAIL
L:   7 | P:   1 |  PRS | Line 7, Position 1: Found unparsable section: 'CREATE OR
                       | ALTER PROCEDURE dbo.IndexOptim...'
WARNING: Parsing errors found and dialect is set to 'tsql'.
```

While this does parse correctly, so there is a work-around.

```sql
CREATE OR ALTER PROCEDURE dbo.DoSomething
AS
BEGIN
    SET NOCOUNT ON;
    SET XACT_ABORT ON;
END
GO
```

### How to reproduce

See provided example with `sqlfluff lint`

### Dialect

tsql

### Version

3.1.0

### Configuration

Defaults, with tsql dialect.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
